### PR TITLE
MINOR: Remove duplicate method in test classes

### PR DESCRIPTION
**Pull request by dengziming**
*More detailed description of your change*
1. Remove duplicate serializing auto-generated data in `RequestConvertToJsonTest`, this is inspired by #9964
2. Remove `RequestTestUtils.serializeRequestWithHeader` since we added a `AbstractRequest.serializeWithHeader` in #10142

*Summary of testing strategy (including rationale)*
Unit test

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by dengziming** [comment]
@chia7712 , PTAL at `RequestConvertToJsonTest` since it's the same with your refactor in #9964;
@abbccdda , I remove `RequestTestUtils.serializeResponseWithHeader` since it's the same with `AbstractRequest.serializeWithHeader` you added in #10142; 
Thank you!

**Comment by chia7712** [comment]
@dengziming Could you merge trunk to trigger QA again? There are some flaky fixes which can stabilize the QA.

**Comment by dengziming** [comment]
`AbstractRequest.serializeWithHeader` will verify apiVersion of request body and header so `ApiVersionsRequestTest.testApiVersionsRequestWithUnsupportedVersion` failed, I just fixed this.

File: core/src/test/scala/unit/kafka/server/BaseRequestTest.scala [hunk 1]
```diff
@@ -136,7 +135,7 @@ abstract class BaseRequestTest extends IntegrationTestHarness {
   }
 
   def sendWithHeader(request: AbstractRequest, header: RequestHeader, socket: Socket): Unit = {
-    val serializedBytes = Utils.toArray(RequestTestUtils.serializeRequestWithHeader(header, request))
+    val serializedBytes = Utils.toArray(request.serializeWithHeader(header))
     sendRequest(socket, serializedBytes)
```
Comment (chia7712) [inline core/src/test/scala/unit/kafka/server/BaseRequestTest.scala:139]: It seems this method can be replaced by `IntegrationTestUtils.sendRequest`

**Review by chia7712 — COMMENTED** [review]
@dengziming thanks for this patch. one small comment below.

File: core/src/test/scala/unit/kafka/server/BaseRequestTest.scala (hunk 1 above)
Comment (dengziming) [reply core/src/test/scala/unit/kafka/server/BaseRequestTest.scala:139]: This is a great suggestion! I even found that the whole `BaseRequestTest` is a duplicate of `IntegrationTestUtils`, how about open another PR for this refactor since this would change a lot of classes and the `IntegrationTestUtils` is under `integration` package while `BaseRequestTest` is under `unit` package.

File: core/src/test/scala/unit/kafka/server/AbstractApiVersionsRequestTest.scala [hunk 2]
```diff
@@ -49,7 +49,9 @@ abstract class AbstractApiVersionsRequestTest(cluster: ClusterInstance) {
     val overrideHeader = IntegrationTestUtils.nextRequestHeader(ApiKeys.API_VERSIONS, Short.MaxValue)
     val socket = IntegrationTestUtils.connect(cluster.brokerSocketServers().asScala.head, cluster.clientListener())
     try {
-      IntegrationTestUtils.sendWithHeader(request, overrideHeader, socket)
+      val serializedBytes = Utils.toArray(
```
Comment (chia7712) [inline core/src/test/scala/unit/kafka/server/AbstractApiVersionsRequestTest.scala:52]: Why this change is necessary? It seems to me the code gets more complicated.

File: core/src/test/scala/unit/kafka/server/BaseRequestTest.scala (hunk 1 above)
Comment (chia7712) [reply core/src/test/scala/unit/kafka/server/BaseRequestTest.scala:139]: sounds good to me

File: core/src/test/scala/unit/kafka/server/AbstractApiVersionsRequestTest.scala (hunk 2 above)
Comment (dengziming) [reply core/src/test/scala/unit/kafka/server/AbstractApiVersionsRequestTest.scala:52]: Because this method is used to test the version inconsistency between request header and request body, but `IntegrationTestUtils.sendWithHeader` will inspect the version and throws `IllegalArgumentException` on inconsistency after this PR.
